crates/router/src/connector/braintree/transformers.rs

```diff
@@ -255,234 +544,1268 @@ impl From<BraintreePaymentStatus> for enums::AttemptStatus {
+                    }
+                }
+            },
+            ..item.data
+        })
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct BraintreeRSyncRequest {
+    query: String,
+}
+
+impl TryFrom<&types::RefundSyncRouterData> for BraintreeRSyncRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::RefundSyncRouterData) -> Result<Self, Self::Error> {
+        let metadata: BraintreeMeta = utils::to_connector_meta_from_secret(
+            item.connector_meta_data.clone(),
+        )
+        .change_context(errors::ConnectorError::InvalidConnectorConfig { config: "metadata" })?;
+        utils::validate_currency(
+            item.request.currency,
+            Some(metadata.merchant_config_currency),
+        )?;
+        let refund_id = item.request.get_connector_refund_id()?;
+        let query = format!("query {{ search {{ refunds(input: {{ id: {{is: \"{}\"}} }}, first: 1) {{ edges {{ node {{ id status createdAt amount {{ value currencyCode }} orderId }} }} }} }} }}",refund_id);
+
+        Ok(Self { query })
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct RSyncNodeData {
+    id: String,
+    status: BraintreeRefundStatus,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct RSyncEdgeData {
+    node: RSyncNodeData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct RefundData {
+    edges: Vec<RSyncEdgeData>,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct RSyncSearchData {
+    refunds: RefundData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct RSyncResponseData {
+    search: RSyncSearchData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct RSyncResponse {
+    data: RSyncResponseData,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum BraintreeRSyncResponse {
+    RSyncResponse(Box<RSyncResponse>),
+    ErrorResponse(Box<ErrorResponse>),
+}
+
+impl TryFrom<types::RefundsResponseRouterData<api::RSync, BraintreeRSyncResponse>>
+    for types::RefundsRouterData<api::RSync>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::RSync, BraintreeRSyncResponse>,
+    ) -> Result<Self, Self::Error> {
+        match item.response {
+            BraintreeRSyncResponse::ErrorResponse(error_response) => Ok(Self {
+                response: build_error_response(&error_response.errors, item.http_code),
+                ..item.data
+            }),
+            BraintreeRSyncResponse::RSyncResponse(rsync_response) => {
+                let edge_data = rsync_response
+                    .data
+                    .search
+                    .refunds
+                    .edges
+                    .first()
+                    .ok_or(errors::ConnectorError::MissingConnectorRefundID)?;
+                let connector_refund_id = &edge_data.node.id;
```

Comment:
Why are we updating the connector_refund_id in the Rsync call?

Reply:
connector_refund_id is a string  , will have to pass it again 
although the refund_id is same only

Reply:
We should not update the refund_id in rsync, please remove this in subsequent PR